app/Services/BannerFinder.php

```diff
@@ -89,42 +85,57 @@ public static function getBestBanners(array $zones, array $keywords)
             foreach ($zones as $zoneInfo) {
                 $zone = Zone::find($zoneInfo['zone']);
 
-                try {
-                    // $zone instanceof Zone; // ?? Yodahack : what the hack
-                    $bannerIds[] = NetworkBanner::where('creative_width', $zone->width)
-                      ->where('creative_height', $zone->height)
-                      ->whereIn('creative_type', $typeDefault)
-                      ->get()->pluck('uuid')->random();
-                } catch (\InvalidArgumentException $e) {
-                    $bannerIds[] = '';
+                if ($zone) {
+                    try {
+                        // $zone instanceof Zone; // ?? Yodahack : what the hack
+                        $bannerIds[] = NetworkBanner::where('creative_width', $zone->width)
+                            ->where('creative_height', $zone->height)
+                            ->whereIn('creative_type', $typeDefault)
+                            ->get()->pluck('uuid')->random();
+                    } catch (\InvalidArgumentException $e) {
+                        $bannerIds[] = '';
+                    }
+                } else {
+                    $bannerIds[] = md5(rand());
                 }
             }
         }
 
         $banners = [];
         foreach ($bannerIds as $bannerId) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "$bannerId" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1481)